extractor/logInterpreter.py

```diff
@@ -144,11 +144,40 @@ def createEmptySessionDict(self, sessionID, time, sipTo, sipFrom):
             "messages": []
         }
     
+
     def checkIfDictKeysContainsRelatedSessions(self, sipHeader):
         pattern = re.compile(self.relatedSessionFormatPattern)
         matchingHeaders = [key for key in sipHeader if pattern.search(key)]
         return matchingHeaders
-    
+
+
+    def filterAssociatedSessions(self, dict, sessionID):
+        result = {}
+        relatedSessions = []
+
+        if sessionID == '':
+            print('No sessionID')
+            return json.dumps(list(dict.values()), indent=2)
+        else:
+            print('SessionID: ', sessionID)
+            # Finds relatedSessions for selected sessionID
+            if sessionID in dict:
+                for key in dict:
+                    if key == sessionID:
+                        relatedSessions = dict[key]['sessionInfo']['associatedSessions']
+                
+                print(f"Related sessions: {relatedSessions}")
+
+                # Filters out all sessions not in relatedSessions[]
+                for x in relatedSessions:
+                    for key in dict:

```

Comment:
As x is a sessionID, would using result = [], then for x in relatedsessions, if (dict[x]) then result.append(dict[x]) work instead of iterating over all keys?